### ISSUE TITLE
Nested loop with _ variables wrongly show overrides warning.

### ISSUE THREAD

**Issue by deniscostadsc** [koalaman/shellcheck#2298]
#### For bugs
- SC2167, SC2165: 
- My shellcheck version:
 ```
ShellCheck - shell script analysis tool
version: 0.7.1
license: GNU General Public License, version 3
website: https://www.shellcheck.net 
```
- [x] https://shellcheck.net/wiki/SC2167, https://shellcheck.net/wiki/SC2165
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

set -euo pipefail

for _ in $(seq 10); do
    for _ in $(seq 10); do
        :
    done
done
```

#### Here's what shellcheck currently says:
```
In ./solutions/uri/2168/generate_in.sh line 5:
for _ in $(seq 10); do
^-^ SC2167: This parent loop has its index variable overridden.


In ./solutions/uri/2168/generate_in.sh line 6:
    for _ in $(seq 10); do
    ^-^ SC2165: This nested loop overrides the index variable of its parent.
```
#### Here's what I wanted or expected to see:

AS [SC2034](https://github.com/koalaman/shellcheck/wiki/SC2034) says, intentionally unused variables should be declared as `_`. I had to declare nested loops, but I didn't want to use the variables created by them, so I did as SC2034 says. But I expected get no warnings, but I got errors above.

**Comment by koalaman** [comment]
You're right, this has now been fixed. Good catch, thanks!
